src/app/pages/services/components/service-nfs/service-nfs.component.ts

```diff
@@ -4,124 +4,170 @@ import * as _ from 'lodash';
+        }
+      ]
     },
     {
-      type: 'checkbox',
-      name: 'nfs_srv_16',
-      placeholder: helptext.nfs_srv_16_placeholder,
-      tooltip: helptext.nfs_srv_16_tooltip,
-      relation: helptext.nfs_srv_16_relation,
+      name: 'divider',
+      divider: true
     },
     {
-      type: 'checkbox',
-      name: 'nfs_srv_mountd_log',
-      placeholder: helptext.nfs_srv_mountd_log_placeholder,
-      tooltip: helptext.nfs_srv_mountd_log_tooltip,
+      name: helptext.nfs_srv_fieldset_other,
+      label: true,
+      config: [
+        {
+          type: 'checkbox',
+          name: 'udp',
+          placeholder: helptext.nfs_srv_udp_placeholder,
+          tooltip: helptext.nfs_srv_udp_tooltip,
+        },
+        {
+          type: 'checkbox',
+          name: 'allow_nonroot',
+          placeholder: helptext.nfs_srv_allow_nonroot_placeholder,
+          tooltip: helptext.nfs_srv_allow_nonroot_tooltip,
+        },
+        
+        {
+          type: 'checkbox',
+          name: 'userd_manage_gids',
+          placeholder: helptext.nfs_srv_16_placeholder,
+          tooltip: helptext.nfs_srv_16_tooltip,
+          relation: helptext.nfs_srv_16_relation,
+        },
+        {
+          type: 'checkbox',
+          name: 'mountd_log',
+          placeholder: helptext.nfs_srv_mountd_log_placeholder,
+          tooltip: helptext.nfs_srv_mountd_log_tooltip,
+        },
+        {
+          type: 'checkbox',
+          name: 'statd_lockd_log',
+          placeholder: helptext.nfs_srv_statd_lockd_log_placeholder,
+          tooltip: helptext.nfs_srv_statd_lockd_log_tooltip,
+        }
+      ]
     },
     {
-      type: 'checkbox',
-      name: 'nfs_srv_statd_lockd_log',
-      placeholder: helptext.nfs_srv_statd_lockd_log_placeholder,
-      tooltip: helptext.nfs_srv_statd_lockd_log_tooltip,
-    },
+      name: 'divider',
+      divider: true
+    }
   ];
 
   private nfs_srv_bindip: any;
   constructor(protected router: Router, protected route: ActivatedRoute,
     protected rest: RestService, protected ws: WebSocketService,
   ) {}
 
-  afterInit(entityForm: any) {
+  afterInit(entityForm: EntityFormComponent) {
+    entityForm.submitFunction = body => this.ws.call('nfs.update', [body]);
+
     this.ws.call('notifier.choices', ['IPChoices']).subscribe((res) => {
-      this.nfs_srv_bindip = _.find(this.fieldConfig, { name: 'nfs_srv_bindip' });
-      for (let item of res) {
+      this.nfs_srv_bindip =
+        _.find(this.fieldSets, { name: helptext.nfs_srv_fieldset_general }).config.find(config => config.name === 'bindip');
+      for (const item of res) {
         this.nfs_srv_bindip.options.push({ label: item[0], value: item[1] });
       }
     });
 
-    entityForm.formGroup.controls['nfs_srv_16'].valueChanges.subscribe((res)=> {
-      if (entityForm.formGroup.controls['nfs_srv_v4'].value) {
+    entityForm.formGroup.controls['userd_manage_gids'].valueChanges.subscribe((res)=> {
```

Comment:
these lines are supposed to enable/disable v4_v3owner but they don't seem to be doing that for some reason.